core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -325,6 +327,11 @@ public Location next() {
         try {
           reader.loadNext();
         } catch (IOException e) {
+          try {
+            reader.close();
+          } catch(IOException e2) {
+            logger.error("Error while closing spill reader", e2);
```

Comment:
@hvanhovell, the problem here is that `Closeables` declares that it throws `IOException`, so even if we tell it to swallow the exception we'd still have to have a `try` block here in order to make the compiler happy.